httpx/_urlparse.py

```diff
@@ -160,7 +160,12 @@ def urlparse(url: str = "", **kwargs: str | None) -> ParseResult:
     # If a URL includes any ASCII control characters including \t, \r, \n,
     # then treat it as invalid.
     if any(char.isascii() and not char.isprintable() for char in url):
-        raise InvalidURL("Invalid non-printable ASCII character in URL")
+        char = [char for char in url if char.isascii() and not char.isprintable()][0]
```

Comment:
Could this be done lazily without materializing the list?

Reply:
It can be done lazily by materializing a generator yes. ✌️